src/CryptoPaymentsSigner.js

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2021 Freeverse.io <dev@freeverse.io>
+// Library for signing API calls to Living Assets market
+
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// 'Software'), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const Contract = require('web3-eth-contract');
+const IERC20JSON = require('./contracts/IERC20.json');
+const PaymentsJSON = require('./contracts/PaymentsERC20.json');
+
+class ERC20Payments {
+  constructor() {
+    this.erc20Contract = {};
+    this.paymentsContract = {};
+  }
+
+  async setupContracts({ paymentsAddr, erc20Addr, provider }) {
+    this.erc20Contract = await new Contract(IERC20JSON.abi, erc20Addr);
```

Comment:
no need for await : Contract do not return a Promise.

why provider ?
```web3.eth.Contract(IERC20JSON.abi, erc20Addr)```

no need to import anything and the user works with, call setupContract({p0, p1, web3}) instead of setupContract({p0, p1, web3.currentProvider})